### PR TITLE
[pickers] Throw error when using adapter from `@date-io`

### PR DESCRIPTION
**Pull request by flaviendelangle**
Need #6625

### CODE REVIEW

**Comment by flaviendelangle** [comment]
Yes, I would prefer to merge #6625 before to make sure the URL is final for v6

**Comment by flaviendelangle** [comment]
Link updated

File: packages/x-date-pickers/src/internals/hooks/useField/useField.ts [hunk 1]
```diff
@@ -34,7 +34,7 @@ export const useField = <
 >(
   params: UseFieldParams<TValue, TDate, TSection, TForwardedProps, TInternalProps>,
 ): UseFieldResponse<TForwardedProps> => {
-  const utils = useUtils<TDate>() as MuiPickerFieldAdapter<TDate>;
+  const utils = useUtils<TDate>() as MuiPickersAdapter<TDate>;
```
Comment (LukasTy) [inline packages/x-date-pickers/src/internals/hooks/useField/useField.ts:37]: It does not seem that this casting is needed anymore 🤔

File: packages/x-date-pickers/src/internals/hooks/useField/useFieldState.ts [hunk 2]
```diff
@@ -39,7 +39,7 @@ export const useFieldState = <
 >(
   params: UseFieldParams<TValue, TDate, TSection, TForwardedProps, TInternalProps>,
 ) => {
-  const utils = useUtils<TDate>() as MuiPickerFieldAdapter<TDate>;
+  const utils = useUtils<TDate>() as MuiPickersAdapter<TDate>;
```
Comment (LukasTy) [inline packages/x-date-pickers/src/internals/hooks/useField/useFieldState.ts:42]: Same with this one

**Comment by flaviendelangle** [comment]
@LukasTy too bad :laughing: 
Will do a quick automerged follow up

**Pull request merged**